docs/spec_draft.md

```diff
@@ -1406,6 +1407,61 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng
+
+### Semantics
+
+Constructs a `result` tensor of a given shape `shape` with random numbers
+generated using the `rng_distribution` algorithm. If `rng_distribution` $=$
+`UNIFORM`, then the random numbers are generated following the uniform
+distribution over the interval [`a`, `b`). If `rng_distribution` $=$ `NORMAL`,
+then the random numbers are generated following the normal distribution with
+mean = `a` and standard deviation = `b`.
+
+If `rng_distribution = UNIFORM` and `b` $\le$ `a`, the result is
+implementation-defined.
```

Comment:
By implementation defined, I mean that if some random number generator has this implementation:
```
rng(a, b, shape):
  result = [...]
  a, b = if a > b ? b , a :  a, b
  if a == b:
    result[:] = None
  ...
  
```
it will be able to handle the a >= b case. If other algorithms returns failure or handles it differently, that could also happen.

> I.e. is there a legitimate use for UNIFORM in conjunction with a  b?

I don't think so.

Reply:
"I don't think so". In that case, let's say:

```
If `rng_distribution = UNIFORM` and `a` $\ge$ `b`, the behavior is undefined.
Same for `rng_distribution = NORMAL` and negative `b`.
```

Undefined behavior is the tightest form of restrictions we can put on runtime semantics, so let's do this if we have a choice (which we seem we have in this situation).